src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSSurfaceNamer.java

```diff
@@ -153,4 +160,90 @@ public String getFieldGetFunctionName(TypeRef type, Name identifier) {
   public String getAsyncApiMethodName(Method method, VisibilityConfig visibility) {
     return getApiMethodName(Name.upperCamel(method.getSimpleName()), visibility);
   }
+
+  @Override
+  public String getLinkedElementName(ProtoElement element) {
+    if (isExternalFile(element.getFile())) {
+      String fullName = element.getFullName();
+      return String.format("[%s]{@link external:\"%s\"}", fullName, fullName);
+    } else {
+      String simpleName = element.getSimpleName();
+      return String.format("[%s]{@link %s}", simpleName, simpleName);
+    }
+  }
+
+  @Override
+  public String getFieldTypeDoc(ModelTypeTable typeTable, Field field) {
+    TypeRef type = field.getType();
+
+    String cardinalityComment = "";
+    if (type.getCardinality() == TypeRef.Cardinality.REPEATED) {
+      if (type.isMap()) {
+        String keyType = getParamTypeName(typeTable, type.getMapKeyField().getType());
+        String valueType = getParamTypeName(typeTable, type.getMapValueField().getType());
+        return String.format("Object.<%s, %s>", keyType, valueType);
+      } else {
+        cardinalityComment = "[]";
+      }
+    }
+    String typeComment = getParamTypeName(typeTable, field.getType());
+    return String.format("%s%s", typeComment, cardinalityComment);
+  }
+
+  private boolean isExternalFile(ProtoFile file) {
+    String filePath = file.getSimpleName();
+    for (String commonPath : COMMON_PROTO_PATHS) {
+      if (filePath.startsWith(commonPath)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public String getParamTypeName(ModelTypeTable typeTable, TypeRef type) {
+    switch (type.getKind()) {
+      case TYPE_MESSAGE:
+        return "Object";
+      case TYPE_ENUM:
+        return "number";
+      default:
+        {
+          String name = PRIMITIVE_TYPE_NAMES.get(type.getKind());
+          if (!Strings.isNullOrEmpty(name)) {
+            return name;
+          }
+          throw new IllegalArgumentException("unknown type kind: " + type.getKind());
+        }
+    }
+  }
+
+  private static final ImmutableSet<String> COMMON_PROTO_PATHS =
+      ImmutableSet.of(
+          "google/api",
+          "google/bytestream",
+          "google/logging/type",
+          "google/longrunning",
+          "google/protobuf",
+          "google/rpc",
+          "google/type");
+
+  private static final ImmutableMap<Type, String> PRIMITIVE_TYPE_NAMES =
```

Comment:
Very similar or same mapping exists in `NodeJSModelTypeNameConverter`. Can we merge it?